core/class_db.cpp

```diff
@@ -275,7 +275,7 @@ bool ClassDB::is_parent_class(const StringName &p_class, const StringName &p_inh
 
 		if (inherits == p_inherits)
 			return true;
-		inherits = get_parent_class(inherits);
```

Comment:
Please add `is_parent_class_nocheck` instead.

Reply:
I mean, assuming that the version with check was used on purpose so far, it's better to add a new version without the check (as done for `get_parent_class` itself) for the one use case that needs it this way. Unless @reduz says it's OK this way, then I'm fine with it.

Reply:
(So let's maybe wait for @reduz's input before doing the change I suggested.)

Reply:
...Welp.

Reply:
Discussed it with @reduz, he says it would likely be better to validate the class name before calling `ClassDB::is_parent_class()` instead of changing ClassDB.

Reply:
You can test with `ClassDB::class_exists(p_doc->name)` before calling `is_parent_class()`.

Reply:
Already on it, just compiling it first to test if everything is working fine.